polyfill/lib/ecmascript.mjs

```diff
@@ -749,10 +749,10 @@ export const ES = ObjectAssign({}, ES2020, {
       if (digits === 'auto') return { precision: 'auto', unit: 'nanosecond', increment: 1 };
       throw new RangeError(`fractionalSecondDigits must be 'auto' or 0 through 9, not ${digits}`);
     }
-    if (NumberIsNaN(digits) || digits < 0 || digits > 9) {
+    const precision = ES.ToInteger(digits);
+    if (!NumberIsFinite(digits) || precision < 0 || precision > 9) {
```

Comment:
Is the NumberIsFinite check needed?

Reply:
It could be `NumberIsNaN` instead, but this aligns better with the spec text.

Reply:
Pretty sure `ToInteger` should return zero for NaN

Reply:
`ES.ToInteger(NaN)` does return 0, but note that the `NumberIsFinite` check is against its _input_ rather than its _output_. The spec text here calls for a RangeError when _fractionalDigitsVal_ is not finite or ToIntegerOrInfinity(_fractionalDigitsVal_) is out of bounds, which is equivalent to this code in the polyfill that throws a RangeError when `digits` is not finite or `ES.ToInteger(digits)` is out of bounds.

Reply:
Done.